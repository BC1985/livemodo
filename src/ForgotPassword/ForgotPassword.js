import React, { Component } from "react";
import { withRouter, Redirect } from "react-router-dom";
// import PasswordConfirmation from './PasswodConfirmation'

class ForgotPassword extends Component {
  state = {
    Redirect: false
  };

  handleSubmit = () => {
    this.setState({
      Redirect: true
    });
  };
  render() {
    if (this.state.Redirect) {
      return <Redirect to="/confirmation" />;
    }
    return (
      <>
        <h3>Forgot your password? No worries...</h3>
        <label>
          Please enter your email in the box below and we'll send you a link to
          reset your password
        </label>
        <form onSubmit={this.handleSubmit}>
          <input type="text" />
          <button>Submit</button>
        </form>
      </>
    );
  }
}

export default withRouter(ForgotPassword);
