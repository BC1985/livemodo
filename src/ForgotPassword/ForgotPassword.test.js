import React from "react";
import ReactDOM from "react-dom";
import { MemoryRouter } from "react-router-dom";
import ForgotPassword from "../ForgotPassword/ForgotPassword";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <MemoryRouter>
      <ForgotPassword />
    </MemoryRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as exptected", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <ForgotPassword />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
