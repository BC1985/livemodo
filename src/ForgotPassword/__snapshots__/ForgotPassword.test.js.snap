// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as exptected 1`] = `
Array [
  <h3>
    Forgot your password? No worries...
  </h3>,
  <label>
    Please enter your email in the box below and we'll send you a link to reset your password
  </label>,
  <form
    onSubmit={[Function]}
  >
    <input
      type="text"
    />
    <button>
      Submit
    </button>
  </form>,
]
`;
